Merge master into develop

sphuber opened the pull request:
This merges in the changes of `v0.12.3`

ltalirz commented:
Looking through the changes now...

ltalirz commented on aiida/manage/fixtures.py:
```diff
@@ -96,19 +96,54 @@ class FixtureManager(object):  # pylint: disable=too-many-public-methods,useless
+        def new_database_with_daemon(aiida_profile):
+            # Configure and start daemon before running JobProcesses
+                #
+                # Note: While the global daemon implementation makes this necessary in aiida-core 0.12.x,
```
I think this can be removed now

sphuber replied:
The whole function `new_database_with_daemon` can be removed correct?

ltalirz replied:
yes

ltalirz commented on aiida/restapi/translator/node.py:
```diff
@@ -606,78 +607,125 @@ def get_node_shape(ntype):
             for node_input in qb_obj.iterdict():
                 node = node_input['in']['*']
-                linktype = node_input['main--in']['label']
+                # add node if it is not present
```
should this be moved a bit down before the "if" statement?

sphuber replied:
done

ltalirz commented on docs/source/datatypes/bands.rst:
```diff
@@ -0,0 +1,116 @@
+To start working with ``BandsData`` we should import it using the
+``DataFactory`` and create an object of type ``BandsData``::
+
+    from aiida.orm import DataFactory
```
```suggestion
    from aiida.plugins import DataFactory
```

sphuber replied:
done

sphuber commented:
@ltalirz was the change in the version number type of the export import necessary here? If so, why wasn't this broken in the release or `provenance_redesign`?

sphuber commented on aiida/transports/plugins/test_all_plugins.py:
```diff
@@ -1263,6 +1263,7 @@ class TestExecuteCommandWait(unittest.TestCase):
     It also checks for escaping of the folder names.
     """
 
+    @unittest.skip('reenable when issue #2460 has been addressed')
```
Yes this was already removed in `provenance_redesign` but not yet in release which re-added it in the merge

sphuber commented:
I have cleaned your two additional commits a bit such that they only include the changes as described in the commit message. Any other changes that are just leftovers that should have gone in the merge commit as well as the additions to address your comments in this PR, I have added to the merge commit

ltalirz commented:
> @ltalirz was the change in the version number type of the export import necessary here? If so, why wasn't this broken in the release or provenance_redesign?

It was complaining that it was comparing float against string

ltalirz commented:
I've also fixed the yaml problem - feel free to rebase if your heart desires it

sphuber commented:
> I've also fixed the yaml problem - feel free to rebase if your heart desires it

Nope, looks great :+1: if the tests finish, I am happy to merge

sphuber commented:
@ltalirz I think your last commit broke the tests, the linter started adding the unicode directives again in the `environment.yml`

ltalirz commented:
I leave the magic merge to you :-)
